### ISSUE TITLE
Haxe: Remove FlexSDK dependency from GUICore compilation

### ISSUE THREAD

**Issue by Igazine** [Moonshine-IDE/Moonshine-IDE#1127]
Remove FlexSDK dependency from GUICore compilation, so the SWC can be built with 

`openfl build flash`

or

`openfl build flash -debug`

**Comment by Igazine** [comment]
Please read our discussion with @joshtynjala [here](https://github.com/Moonshine-IDE/Moonshine-IDE/commit/7f7f1dde78e143b340fe0523f4a77fab438a5ccf) regarding this issue.

**Comment by piotrzarzycki21** [comment]
I'm getting following error -> 

```
 openfl build air -debug
Error:  Library not found : flexhome/frameworks/libs/framework.swc
```

is this related to latest changes ?

**Comment by Igazine** [comment]
This currently a work-in-progress in features/haxedevelop branch, I'll mark the issue test ready when it's done and merged into master.

**Comment by piotrzarzycki21** [comment]
This blocks me from moving forward, because I have following issue: 

```
s/impls/IAboutBridgeImp.as:30:38
[MoonshineDESKTOPevolved]: Interface IAboutBridge was not found.
```

It's probably means that I need to rebuild MoonshineGUICore - where build is failing.

**Comment by joshtynjala** [comment]
@piotrzarzycki21 Until it gets merged, you need to add `-Dflexhome=path/to/flex-sdk` to your build command (with the path to the same Flex SDK you use to build Moonshine).

**Comment by Igazine** [comment]
Please try building the swc like this:

`openfl build flash -Dflexhome={path-to-flex-home-dir} -Dair -debug`

or run 

`ant CompileGUICore` from

MoonshineDESKTOPevolved/build

**Comment by joshtynjala** [comment]
> openfl build flash -Dflexhome={path-to-flex-home-dir} -Dair -debug

Note: yesterday, I made it so that you can use the `air` target in the master branch, so a slightly simpler command is `openfl build air -debug -Dflexhome={path-to-flex-home-dir}`

**Comment by piotrzarzycki21** [comment]
```
openfl build air -debug -Dflexhome=/Users/piotrzarzycki/Downloads/MoonshineSDKs/Flex_SDK/Flex_4.16.1_AIR_33.1
Error: You must define AIR_SDK with the path to your AIR SDK
```
Why I get that actually ?

**Comment by joshtynjala** [comment]
> Error: You must define AIR_SDK with the path to your AIR SDK

You have two options:

1) Run `lime config AIR_SDK path/to/air-sdk` to configure which AIR SDK OpenFL/Lime should use when building for the air target. This will be used for all projects that target air.

2) Add `-DAIR_SDK=path/to/air-sdk` to your build command too. It can be the same Flex SDK that you define with `-Dflexhome`. This would need to be done every time that you build.

I tend to prefer option 1, since I can set it and forget it. 😄

**Comment by piotrzarzycki21** [comment]
> This blocks me from moving forward, because I have following issue:
> 
> ```
> s/impls/IAboutBridgeImp.as:30:38
> [MoonshineDESKTOPevolved]: Interface IAboutBridge was not found.
> ```
> 
> It's probably means that I need to rebuild MoonshineGUICore - where build is failing.

I was able to build _MoonshineGUICore_ and I'm still getting this issue. I was thinking that maybe I need to build as well _MoonshineSDKInstallerGUICore_ - I'm getting following error while trying:

```
openfl build air -debug -Dflexhome=/Users/piotrzarzycki/Downloads/MoonshineSDKs/Flex_SDK/Flex_4.16.1_AIR_33.1 -DAIR_SDK=/Users/piotrzarzycki/Downloads/MoonshineSDKs/Flex_SDK/Flex_4.16.1_AIR_33.1
bin/air/haxe/ApplicationMain.hx:288: characters 48-52 : Type not found : Main
```

**Comment by piotrzarzycki21** [comment]
> > This blocks me from moving forward, because I have following issue:
> > ```
> > s/impls/IAboutBridgeImp.as:30:38
> > [MoonshineDESKTOPevolved]: Interface IAboutBridge was not found.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It's probably means that I need to rebuild MoonshineGUICore - where build is failing.
> 
> I was able to build _MoonshineGUICore_ and I'm still getting this issue. I was thinking that maybe I need to build as well _MoonshineSDKInstallerGUICore_ - I'm getting following error while trying:
> 
> ```
> openfl build air -debug -Dflexhome=/Users/piotrzarzycki/Downloads/MoonshineSDKs/Flex_SDK/Flex_4.16.1_AIR_33.1 -DAIR_SDK=/Users/piotrzarzycki/Downloads/MoonshineSDKs/Flex_SDK/Flex_4.16.1_AIR_33.1
> bin/air/haxe/ApplicationMain.hx:288: characters 48-52 : Type not found : Main
> ```

We have connected with @Igazine over a call and this issue on my sight is resolved. Thank you!

**Comment by Igazine** [comment]
The SWC can be build with 

`openfl build flash`

command (with optional `-debug`)

Changes have been merged into master

**Comment by joshtynjala** [comment]
On master, last week, I made is so that you can now use `openfl build air -debug` to build the SWC.

**Comment by joshtynjala** [comment]
Or maybe @Igazine reverted my changes? I hope not! 70569a390002f98328b93c0b177a9a19f22916b5

**Comment by Igazine** [comment]
Some team members had several issues compiling the SWC targeting AIR. Configuring AIR SDK actually raised more questions and issues than it had answered. This is currently the cleanest way the SWC should be built.

I'm not going to touch this issue anymore, as it takes up way too much of my time, and this step will be entirely removed once the conversion is complete.

**Comment by joshtynjala** [comment]
> Some team members had several issues compiling the SWC targeting AIR.

Anyone have a link to where those issues were reported?

**Comment by piotrzarzycki21** [comment]
> > Some team members had several issues compiling the SWC targeting AIR.
> 
> Anyone have a link to where those issues were reported?

It was probably me who has problems with understanding whole process. Earlier I just could build with command in instructions, later it's started to fail etc.
